### PR TITLE
Test generation of struct-related bytecodes and bytecode sequences.

### PR DESCRIPTION
**Pull request by tzakian**
Allow generation of structs  (so `Pack` and `Unpack` instructions) in test generation. Only look at the top commit on this PR. 

Needed to fix an off-by-1 error in the stack usage accounting for `Pack` that was leading to unbalanced stack usage errors.

### CODE REVIEW

File: language/tools/test-generation/src/abstract_state.rs [hunk 1]
```diff
@@ -42,6 +42,7 @@ pub enum Mutability {
 impl AbstractValue {
     /// Create a new primitive `AbstractValue` given its type; the kind will be `Unrestricted`
     pub fn new_primitive(token: SignatureToken) -> AbstractValue {
+        debug!("new prim {:#?}", token);
```
Comment (shazqadeer) [inline language/tools/test-generation/src/abstract_state.rs]: Did you mean to leave this debug statement in the code?

File: language/tools/test-generation/src/bytecode_generator.rs [hunk 2]
```diff
@@ -491,17 +492,17 @@ impl BytecodeGenerator {
...
-                        SignatureToken::Bool => Bytecode::LdFalse,
-                        SignatureToken::ByteArray => {
-                            Bytecode::LdByteArray(ByteArrayPoolIndex::new(0))
-                        }
-                        _ => unimplemented!("Unsupported return type: {:#?}", abstract_value.token),
-                    };
-                    state = self.apply_instruction(state, &mut bytecode, next_instruction);
+                    let next_instructions =
+                        inhabit_with_bytecode_seq(&module, &abstract_value.token);
+                    debug!(
```
Comment (shazqadeer) [inline language/tools/test-generation/src/bytecode_generator.rs]: Same comment here.

File: language/tools/test-generation/src/transitions.rs [hunk 3]
```diff
@@ -176,7 +176,7 @@ pub fn stack_satisfies_struct_signature(
         .flatten()
         .map(|field| field.type_signature().token());
     let mut satisfied = true;
-    for (i, token_view) in field_token_views.enumerate() {
+    for (i, token_view) in field_token_views.rev().enumerate() {
```
Comment (shazqadeer) [inline language/tools/test-generation/src/transitions.rs]: Was this a bug in Darion's code where he got the order in which values are unpacked from the stack wrong?

File: language/tools/utils/src/inhabitation/inhabit.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use vm::file_format::{
+    AddressPoolIndex, ByteArrayPoolIndex, Bytecode, CompiledModuleMut, LocalsSignatureIndex,
+    SignatureToken, StructDefinitionIndex, StructFieldInformation, TableIndex, UserStringIndex,
+};
+
+// Generate a sequence of instructions that inhabits the value of `token`
```
Comment (shazqadeer) [inline language/tools/utils/src/inhabitation/inhabit.rs]: I believe the more precise description of the behavior of this function (based on our offline discussion) is as follows:

Generate a sequence of instructions whose overall effect is to push a single value of type `token` on the stack, specifically without consuming any values that existed on the stack prior to the execution of the instruction sequence.

If you agree with my description, please incorporate it in the comment.

File: language/tools/test-generation/src/transitions.rs (hunk 3 above)
Comment (tzakian) [reply language/tools/test-generation/src/transitions.rs]: Yes. This was leading to us trying to pack in the types in the reverse order.

**Pull request closed without merging**